src/components/topNav/primaryTopNav/PrimaryTopNav.tsx

```diff
@@ -49,24 +64,43 @@ export const PrimaryTopNav = ({ items, mobileLogo, layoutClassName }: TopNavProp
         <ul className={styles.ul}>
           {items.map(({ label, icon, current, handleClick, subItems }, idx) => (
             <li
-              onClick={() => handleItemClick(handleClick)}
+              onClick={() =>
+                !subItems
+                  ? handleItemClick(handleClick)
+                  : screenWidth > 992
+                  ? handleItemClick(handleClick)
+                  : handleSubItemMenu(idx, !subNavIsOpen[idx as keyof typeof subNavIsOpen] ?? true)
+              }
               className={clsx(styles.li, current && styles.current)}
               key={idx}
             >
-              <Link className={clsx(styles.link, styles.label)}>
+              <Link
+                className={clsx(
+                  styles.link,
+                  styles.label,
+                  subItems && styles.mobileLink,
+                  current && styles.currentLink,
+                )}
+              >
                 {icon}
-                {label}
+                {label}{" "}
+                {subItems && screenWidth < 992 && (
+                  <FontAwesomeIcon
+                    className={clsx(styles.toggleIcon, subNavIsOpen[idx as keyof typeof subNavIsOpen] && styles.isOpen)}
+                    icon={faChevronRight}
+                  />
+                )}
```

Comment:
I find this a bit complex, see my other comment as well.

Is this because a top-level item can have a link **and** have sub-items?